FileManager/ViewControllers/FilesViewController/FilesViewController.swift

```diff
@@ -263,3 +260,78 @@ private extension FilesViewController {
     }
 }
 
+// MARK: AuthorizationServiceDelegate
+
+extension FilesViewController: AuthorizationServiceDelegate {
+    func handlePasswordIDAutorization(handler: @escaping (String?) -> Void) {
+        let alertController = UIAlertController(title: "Enter password",
+                                                message: "To work with application, enter the application password",
+                                                preferredStyle: .alert)
+        
+        alertController.addAction(UIAlertAction(title: "OK",
+                                                style: .default) { _ in
+            guard let textFields = alertController.textFields else { return }
+
+            if let textField = textFields.first {
+                handler(textField.text)
+            }
+        })
+        
+        alertController.addAction(UIAlertAction(title: "Cancel", style: .cancel))
+        
+        alertController.addTextField { textField in
+            textField.isSecureTextEntry = true
+            textField.placeholder = "Password"
+            
+        }
+        
+        self.present(alertController, animated: true)
+    }
+    
+    func handlePasswordSetting(handler: @escaping (String) -> Void) {
+        let alertController = UIAlertController(title: "Set password",
+                                                message: "To work with application, set the application password",
+                                                preferredStyle: .alert)
+        
+        alertController.addAction(UIAlertAction(title: "Save",
+                                                style: .default) { _ in
+            guard let textFields = alertController.textFields else { return }
+            
+            if let textField = textFields.first, let text = textField.text {
+                handler(text)
+            }
+        })
+        
+        alertController.addTextField { textField in
+            textField.isSecureTextEntry = true
+            textField.placeholder = "Password"
+            textField.addTarget(self, action: #selector(self.textChanged), for: .editingChanged)
+            
+        }
+        alertController.addTextField { textField in
+            textField.isSecureTextEntry = true
+            textField.placeholder = "Confirm"
+            textField.addTarget(self, action: #selector(self.textChanged), for: .editingChanged)
+        }
+        
+        alertController.actions[0].isEnabled = false
```

Comment:
I believe you can use `alertController.actions.first?.isEnabled = false`. Using `.first` is more secure than `[0]`